DataStructuresMemories/ViewControllers/DataStructViewController.swift

```diff
@@ -8,49 +8,103 @@
+            let priorityForUnShrink = isTextShrinked ? UILayoutPriority(999) : UILayoutPriority(250)
+            UIView.transition(
+                with: childView,
+                duration: 0.3,
+                options: [.transitionFlipFromTop],
+                animations: { [ weak self ]  in
+                    self?.toggleButton.setTitleForAllStates(using: title)
+                    self?.wikiButton.isHidden = !(self?.wikiButton.isHidden)!
+                    self?.gradientView.isHidden = !(self?.gradientView.isHidden)!
+                    self?.constraintToHeight.priority = priorityForUnShrink
+                    self?.gradientView.updateConstraints()
+                }
+            )
+            trailingToggleButtonConstraint.priority = priorityForShrink
+            leadingButtonConstraint.priority = priorityForShrink
+            centreXToggleButtonConstraint.priority = priorityForUnShrink
+        }
+    }
+    @IBAction func touchWikiButton(_ sender: UIButton) {
+        presentWaysToOpenLink()
+    }
     
     @IBAction func touchMoreButton(_ sender: UIButton) {
-        print("more")
-        gradientView.isHidden = true
-        lessInformationButton.isHidden = false
-        constraintToHeight.priority = UILayoutPriority(250)
-//        parentView.co
+        isTextShrinked = !isTextShrinked
     }
     
-    @IBOutlet weak var heightConstrains: NSLayoutConstraint!
+    @IBOutlet weak var trailingToggleButtonConstraint: NSLayoutConstraint!
+    
+    @IBOutlet weak var leadingButtonConstraint: NSLayoutConstraint!
+    
+    @IBOutlet weak var centreXToggleButtonConstraint: NSLayoutConstraint!
+    
+    @IBAction func tappedVisualisationButton(_ sender: UIButton) {
+        pushVisulizationControllerInStack()
+    }
+    @IBOutlet weak var scrollView: UIScrollView!
     
     @IBOutlet weak var childView: UIView!
+    
     @IBOutlet var parentView: UIView!
     
-    @IBOutlet weak var moreInformationButton: UIButton!
+    @IBOutlet weak var toggleButton: UIButton!
     
     @IBOutlet weak var constraintToHeight: NSLayoutConstraint!
+
+    @IBOutlet weak var wikiButton: UIButton!
     
-    @IBOutlet weak var lessInformationButton: UIButton!
-    
-    @IBAction func touchLessButton(_ sender: UIButton) {
-        print(gradientView.isHidden)
-        gradientView.isHidden = false
-        lessInformationButton.isHidden = true
+    func presentWaysToOpenLink() {
+        let waysToOpenWikiLink = UIAlertController(
+            title: "Links To Wikipedia",
+            message: "Choose the way to open wikipedia page with more information.",
+            preferredStyle: .actionSheet
+        )
+        
+        let presenter = URLPresenterManager()
         
-        constraintToHeight.priority = UILayoutPriority(999)
+        for index in 0 ..< presenter.amountOfPresenters {
+            presenter.setPresenter(by: index)
+            let cuurentPresenter = presenter.presenter!
```

Comment:
`presenter.presenter`? Really?

Reply:
I've already changed it to more presentable names.